notebooks/gotcha.clj

```diff
@@ -0,0 +1,27 @@
+(ns gotcha
+  [:require
+   [scicloj.kindly.v4.kind :as k]])
+
+(k/md "# Gotchas...
+Although we try to avoid such things, sometimes, when you're fighting the host language, it's just not practical to carry over the original conventions. Here we will try to keep an up-to-date list of possible surprises (that are not easy to 'fix').
+
+**^** (`Power`) - Despite allowing a wider range of characters, we cannot use `^` in Wolframite because it is reserved for adding metadata to virtually all Clojure symbols. `(^ x 2)` will probably not do what you expect!
+
+**:=** (`SetDelayed`) - Is actually `_=` in Wolframite. This is because `:` is usually a reserved in Clojure for creating keywords. It's worth noting though that this is possibly nicer, in a way, because `_` looks like a placeholder, and defining a placeholder for the expression is what it does. It also sort of implies a delay...
+
+**:>** (`RuleDelayed`) - Is `_>` in Wolframite.
+
+**.** (`Dot`) - Has been changed to `<*>`, in the spirit of the inner product, because `.` is a key character in Clojure, and most object-oriented systems, for namespacing.
+
+**/.** (`ReplaceAll`) -  Similarly, has been changed to `x>>` (and `Replace` to `x>`) for consistency.
+
+**=.** (`Unset`) -  has also been changed to `=!`; be careful not to confuse this with `!=`!
+
+Symbols within threading macros. - After spending so much time playing with symbols, be careful of slipping into things like
+`(-> 'x '(Power 1))`
+This will not work because threads are macros and so combining them with unevaluated functions will lead to errors.
+
+*Symbols passed to Wolfram must be alphanumeric* - In the end, when they get passed to the Wolfram kernel, symbols must be strictly alphanumeric (apart from forward slashes and dollar signs), *i.e.* `r_2` is currently not allowed. This is due to underlying limitations of the Wolfram language. Much like with Mathematica however, we can get around this is general by using Wolframite's aliasing system (see the relevant tutorials).
+
+*Vectors, `[]`, vs lists, `()`.* - Lists are used to represent functions and so when combining Clojure and Wolfram expressions, make sure that data literals are vectors. For example, `(wh/view (w/ListLinePlot (range 10)))` will fail (otherwise unexpectedly), but `(wh/view (w/ListLinePlot (into [] (range 10))))` will give you what you expect.
```

Comment:
```suggestion
*Vectors, `[]`, vs lists, `()`.* - Lists are used to represent function calls and so when combining Clojure and Wolfram expressions, make sure that data literals are vectors. For example, `(wh/view (w/ListLinePlot (range 10)))` will fail (otherwise unexpectedly), but `(wh/view (w/ListLinePlot (into [] (range 10))))` will give you what you expect.
```

How does this fail? I seem getting the same result in both cases. Perhaps simplify and make the two cases use `(1 2 3 4 5)` vs `[1 2 3 4 5]`?

Reply:
Interesting, it was failing for me. I'll need to check if it's a browser thing.

Reply:
![2024-08-02-112553_3840x1080_scrot](https://github.com/user-attachments/assets/62ef8f8a-fdd0-4cd2-a3b4-becdfe232fac)

This definitely doesn't work for me in chromium or firefox. It doesn't raise an error, but the graphic is not displayed.